src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -201,6 +201,7 @@ def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.devic
             num_inference_steps (`int`):
                 the number of diffusion steps used when generating samples with a pre-trained model.
         """
+        num_inference_steps = min(self.config.num_train_timesteps, num_inference_steps)
```

Comment:
Here, I think it's actually better to throw an error if `num_inference_steps > self.config.num_train_timesteps` . It's usually better to educate the user instead of silently correcting a bug IMO. 

More specifically, it's just wrong to pass `num_inference_steps > self.config.num_train_timesteps` since it's impossible for the model to handle it => so the user should understand this and correct the input.

```suggestion
        if num_inference_steps > self.config.num_train_timesteps:
            raise ValueError(f"`num_inference_steps`: {num_inference_steps} cannot be larger than `self.config.train_timesteps`: {self.config.num_train_steps} as the unet model trained with this scheduler can only handle maximal {self.config.num_train_steps} timesteps.")
        num_inference_steps = min(self.config.num_train_timesteps, num_inference_steps)
```

Reply:
Yep. I agree that it's better to throw an error. I'll add the suggested changes.